src/c4/yml/tree.cpp

```diff
@@ -331,25 +321,25 @@ Tree::~Tree()
 }
 
 
-Tree::Tree(Tree const& that) noexcept : Tree(that.m_callbacks)
+Tree::Tree(Tree const& that) : Tree(that.m_callbacks)
 {
     _copy(that);
 }
 
-Tree& Tree::operator= (Tree const& that) noexcept
+Tree& Tree::operator= (Tree const& that)
 {
     _free();
     m_callbacks = that.m_callbacks;
     _copy(that);
     return *this;
 }
 
-Tree::Tree(Tree && that) noexcept : Tree(that.m_callbacks)
+Tree::Tree(Tree && that) : Tree(that.m_callbacks)
 {
     _move(that);
 }
 
-Tree& Tree::operator= (Tree && that) noexcept
```

Comment:
Would `RYML_NOEXCEPT` not work in here (and the rest of the constructors and operator=?

If `RYML_USE_ASSERT` is false, there will be no assert and therefore no exception, so the compiler can use choose move operator more easily.

Reply:
Good point. Indeed, ryml is only asserting inside these functions so RYML_NOEXCEPT could be used.

However if the user provides throwing allocation/free callbacks, then the noexcept would act as a barrier. So for now I think it's best to keep these without RYML_NOEXCEPT.